dwl.c

```diff
@@ -271,10 +281,15 @@ applyrules(Client *c)
 
 	/* rule matching */
 	c->isfloating = 0;
-	if (!(appid = c->xdg_surface->toplevel->app_id))
-		appid = broken;
-	if (!(title = c->xdg_surface->toplevel->title))
-		title = broken;
+	if (c->isxdg) {
+		if (!(appid = c->xdg_surface->toplevel->app_id))
+			appid = broken;
+		if (!(title = c->xdg_surface->toplevel->title))
+			title = broken;
+	} else {
+		if (!(title = c->xwayland_surface->title))
+			title = broken;
```

Comment:
xwayland does not have an app_id

Reply:
It does appear to still have instance/class though

Reply:
> It does appear to still have instance/class though

We can add those when we use them e.g. rules based on class.